src/chatbot/bot.py

```diff
@@ -68,13 +79,170 @@ async def lmgtfy(ctx: Context) -> None:
+                discord_id=ctx.message.author.id
+            )
+            prediction_choice = await PredictionChoice.objects.async_get(
+                choice__icontains=choice
+            )
+
+            if amount > user.gam_coins:
+                logger.info(
+                    "User %d does not have enough GamCoins to wager (tried wagering %d, only had %d)",
+                    user.discord_id,
+                    amount,
+                    user.gam_coins,
+                )
+                await ctx.message.add_reaction(settings.WAGER_ERROR_REACTION)
+                return
+
+            user.gam_coins -= amount
+            await user.async_save()
+
+            wager = Wager(bettor=user, amount=amount, choice=prediction_choice)
+
+            await wager.async_save()
+            await ctx.message.add_reaction(settings.WAGER_SUCCESS_REACTION)
+        except Prediction.DoesNotExist:
+            # This should only happen if someone replies to the wrong message
+            logger.info(
+                "User tried wagering on thread_id %d which does not correspond to a prediction",
+                thread_id,
+            )
+            await ctx.message.add_reaction(settings.WAGER_ERROR_REACTION)
+        except PredictionChoice.DoesNotExist:
+            logger.info(
+                "User's provided choice %s did not match any choices available for prediction %d",
+                choice,
+                prediction.id,
+            )
+            await ctx.message.add_reaction(settings.WAGER_ERROR_REACTION)
+    else:
+        logger.info("User tried to wager without replying to a prediction thread.")
+
+
+@bot.command()
+async def close(ctx: Context) -> None:
+    if ctx.message.reference:
+        thread_id = ctx.message.reference.message_id
+        try:
+            prediction = await Prediction.objects.async_get(thread_id=thread_id)
+            prediction.open = False
+            await prediction.async_save()
+        except Prediction.DoesNotExist:
+            # This should only happen if someone replies to the wrong message
+            logger.info(
+                "User tried closing on thread_id %d which does not correspond to a prediction",
+                thread_id,
+            )
+    else:
+        logger.info(
+            "User tried to close prediction without replying to a prediction thread."
+        )
+
+
+@bot.command()
+async def resolve(ctx: Context, correct_choice: str) -> None:
+    # Resolves a prediction thread by getting all users that made the correct choice
+    # and then assigning a payout based on their relative pot contributions
+    if ctx.message.reference:
+        thread_id = ctx.message.reference.message_id
+        try:
+            prediction = await Prediction.objects.async_get(thread_id=thread_id)
+            correct_prediction_coice = await PredictionChoice.objects.async_get(
+                prediction=prediction, choice__icontains=correct_choice
```

Comment:
`icontains` could be weird here - we probably just want to do `.lower()` on the strings always instead of relying on `icontains`.